inquire/src/prompts/path_select/prompt.rs

```diff
@@ -0,0 +1,576 @@
+        backend.render_options(page, checked)?;
+
+        if let Some(help) = help_message {
+            backend.render_help_message(help)?;
+        }
+        Ok(())
+    }
+
+    fn handle(
+        &mut self,
+        action: PathSelectPromptAction,
+    ) -> InquireResult<crate::prompts::prompt::ActionResult> {
+        let Self {
+            config:
+                PathSelectConfig {
+                    ref page_size,
+                    ref keep_filter,
+                    ..
+                },
+            ref options,
+            ref select_multiple,
+            ref selection_mode,
+            ref mut cursor_index,
+            ref mut checked,
+            ref mut input,
+            ref mut selected,
+            ref mut filtered_options,
+            ref mut current_path,
+            ref mut data_needs_refresh,
+            ref mut sorting_mode,
+            ..
+        } = self;
+        let result = match action {
+            PathSelectPromptAction::MoveUp => self.move_cursor_up(1, true),
+            PathSelectPromptAction::PageUp => self.move_cursor_up(*page_size, true),
+            PathSelectPromptAction::MoveToStart => self.move_cursor_up(usize::MAX, true),
+
+            PathSelectPromptAction::MoveDown => self.move_cursor_down(1, true),
+            PathSelectPromptAction::PageDown => self.move_cursor_down(*page_size, true),
+            PathSelectPromptAction::MoveToEnd => self.move_cursor_down(usize::MAX, true),
+            PathSelectPromptAction::ChangeSortMode => {
+                *sorting_mode = (*sorting_mode).next();
+                *data_needs_refresh = true;
+                ActionResult::NeedsRedraw 
+            }
+            PathSelectPromptAction::ToggleCurrentOption => self.toggle_cursor_selection(),
+            PathSelectPromptAction::NavigateDeeper => {
+                match filtered_options.get(*cursor_index) {
+                    Some(option_index) => match options.get(*option_index) {
+                        Some(PathEntry {
+                            file_type, path, ..
+                        }) if file_type.is_dir() => {
+                            *current_path = path.to_path_buf();
+                            *data_needs_refresh = true;
+                            *cursor_index = 0;
+                            ActionResult::NeedsRedraw
+                        }
+                        _ => ActionResult::Clean,
+                    },
+                    _ => {
+                        /* It might be an empty directory*/
+                        ActionResult::Clean
+                    }
+                }
+            }
+
+            PathSelectPromptAction::NavigateHigher => {
+                if let Some(parent) = current_path.parent() {
+                    *current_path = parent.to_path_buf();
+                    *data_needs_refresh = true;
+                    *cursor_index = 0;
+                }
```

Comment:
I got this error when going up and up, not sure why as it seems you put the handling into place, but worth checking out
```
IO(
    Os {
        code: 2,
        kind: NotFound,
        message: "No such file or directory",
    },
)
```

Reply:
I am not able to replicate this on Ubuntu Linux, but it was indeed a persistent problem before. Can you please let me know what OS you were on so I can try to write a test or something and figure it out?